database/db.go

```diff
@@ -47,22 +47,58 @@ func QueryAllVoters() ([]durn.Voter, error) {
 		return nil, err
 	}
 	defer rows.Close()
-	var res []durn.Voter
+	var res []models.Voter
 
 	for rows.Next() {
-		var username durn.Voter
+		var username models.Voter
 		err = rows.Scan(&username)
 		res = append(res, username)
 	}
 
 	return res, nil
 }
 
-func InsertElection(e durn.Election) error {
+// InsertVoters inserts provided voters into the database, but first queries the database
+// to check if they are already added, in which case they are skipped
+func InsertVoters(voters []models.Voter) error {
+	mutex.Lock()
+	defer mutex.Unlock()
+
+	for _, voter := range voters {
+		var email string
+		err := db.QueryRow(`SELECT * FROM valid_voters WHERE email = $1`, voter).Scan(&email)
+		if err != sql.ErrNoRows {
+			continue
+		}
+		_, err = db.Exec(`INSERT INTO valid_voters VALUES ($1)`, voter)
```

Comment:
Seems inefficient to make two queries for every entry in voters. I guess you could get away with two queries in total by first querying the emails of every voter and adding them to a look-up table. I don't know if this is quicker and it makes little difference since we don't need great performance from this system.

Reply:
I agree regarding querying all voters at first, that seems very reasonable. 

When implementing this, I googled a bit to see if it was possible to do something for inserting, but the result I got was that it seemed hard to do for a variable ammount of values. However, upon closer inspection, it seems you can do variadic things in golang, which makes it fairly simple. So I'll reimplement it this way

Reply:
I tried to implement this using variadics, but the compiler complained (seems to have to do with `Exec` taking `[]interface{}...` as a paramater, rather than a explicit type). So I opted to keep inserting voters one at a time.